bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -81,7 +80,8 @@ public class IpAddonFinder extends BaseAddonFinder {
     private static final String PARAMETER_TIMEOUT_MS = "timeoutMs";
 
     private final Logger logger = LoggerFactory.getLogger(IpAddonFinder.class);
-    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(SERVICE_NAME);
```

Comment:
This created a dedicated pool of 5 threads, although there is a single job to be run only a single time.